mockgcp: create tool for openapi -> proto

justinsb opened the pull request:
A few APIs don't have public protos, and rather than change how we
implement our mocks, it is possible to create a proto from the
published OpenAPI documents.

This is a "jig" for creating mocks, and certainly is not in the
"serving path", so there are going to be special cases and incomplete
coverage in this tool.

jingyih commented:
Mock failure seems unrelated to the PR.

```
--- FAIL: TestAllInSeries/samples/project-level-policy (88.89s)
```

jingyih reviewed (commented):
/lgtm
/approve

I did a quick pass. I am in favor of merging it so people can start using it and improve the tool if they found more edge cases.

anhdle-sso commented:
/lgtm

justinsb commented:
> I am in favor of merging it so people can start using it and improve the tool if they found more edge cases.

Thanks, I think that's a good approach.  It seems to work pretty well for the services I'm encountering so far.  It can obviously use a code cleanup, and probably there are still many unhandled cases, but they've been pretty small tweaks for the last few services I've done (and hopefully there aren't _that_ many more!)

I squashed and rebased, and removed the storage API (that should go in with the PR where we actually mock the storage API)

jingyih commented on mockgcp/apis/mockgcp/storage/v1/service.proto:
```diff
@@ -1,63 +0,0 @@
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-syntax = "proto3";
```
I don't follow why storage v1 service proto was removed? I thought it should be generated by this new tool?

justinsb replied:
So I don't think it belongs in _this_ PR - this PR is just for generation.  I figured I would add it alongside the GCS mockgcp support (which I believe we don't have today).

If you think it would be good to include here (e.g. as an example), then I can add it back in...

jingyih replied:
Oh I see. So it is just because we didn't run `make generate-protos-from-openapi` in this PR?

justinsb replied:
Right!

jingyih commented:
/lgtm
/approve

jingyih commented:
@justinsb, It seems there is some issue in mockgcp proto generation for "storage/v1" after this PR merge.

To reproduce:
```
cd mockgcp
make generate-protos-from-openapi
make gen-proto
```

As the result, 3 files under "mockgcp/generated/mockgcp/storage/v1" will be override to have "package storagepb"; while the 2 remaining files still have "package storage". Now we have 2 different golang packages under the same folder.

jingyih commented:
I created #1563 to temporarily unblock other mockgcp PRs.
